config/runtime.exs

```diff
@@ -94,90 +94,122 @@ config :teiserver, TeiserverWeb.Endpoint,
       endpoint_defaults[:secret_key_base]
     )
 
-use_tls? = Teiserver.ConfigHelpers.get_env("TEI_TLS_PRIVATE_KEY_PATH", nil) != nil
+spring_listeners =
+  Application.get_env(:teiserver, Teiserver.SpringTcpServer)
+  |> Keyword.fetch!(:listeners)
+
+use_tls? =
+  Teiserver.ConfigHelpers.get_env("TEI_TLS_PRIVATE_KEY_PATH", nil) != nil
+
 config :teiserver, Teiserver, use_tls?: use_tls?
 
 tei_defaults = Application.get_env(:teiserver, Teiserver)
 
+spring_listeners =
+  Application.get_env(:teiserver, Teiserver.SpringTcpServer)
+  |> Keyword.fetch!(:listeners)
+
+spring_listeners =
+  if Keyword.get(spring_listeners, :tcp) do
+    # If TCP listener is enabled put the port option
+    get_and_update_in(spring_listeners, [:tcp, :socket_opts, :port], fn default ->
+      {default, Teiserver.ConfigHelpers.get_env("TEI_SPRING_TCP_PORT", default, :int)}
```

Comment:
The runtime configuration logic ended up becoming very convoluted, but it was already convoluted to start with, we had to make the adjustments.

Reply:
Assuming the provisioning uses the environment variables defined [here](https://github.com/beyond-all-reason/ansible-teiserver/blob/7ca4917c65cc8ebe9a52443d6b77c2477a7a4ae0/roles/teiserver/templates/teiserver.env.j2#L3) and nowhere else I identified the body of work necessary.

If you don't mind I'd rather send it as a separate PR.

Reply:
> Assuming the provisioning uses the environment variables defined [here](https://github.com/beyond-all-reason/ansible-teiserver/blob/7ca4917c65cc8ebe9a52443d6b77c2477a7a4ae0/roles/teiserver/templates/teiserver.env.j2#L3) and nowhere else I identified the body of work necessary.
> 
> If you don't mind I'd rather send it as a separate PR.

That's right.
Separate PR is good.